File: designpatterns/src/com/renho/behavior/state/ConcreteStateA.java
package com.renho.behavior.state;

public class ConcreteStateA extends State{

	private int status;
	
	public ConcreteStateA(int status) {
		this.status = status;
	}
	
	@Override
	public void handle(Context context) {
		context.setState(new ConcreteStateB(status + 6));
		System.out.println("ÉÏÎç:" + status);
	}

	public int getStatus() {
		return status;
	}

	public void setStatus(int status) {
		this.status = status;
	}

}

File: ideaworkspace/designpatterns/src/main/scala/com/renho/behavior/state/ConcreteStateA.scala
package com.renho.behavior.state

/**
  * Created by renho on 2016/7/11.
  */
class ConcreteStateA(var status: Int) extends State {
	override def handle(context: Context): Unit = {
		println("上午:" + status)
		context.state = new ConcreteStateB(status + 6)
	}
}
